### ISSUE TITLE
UDFs broken with DuckDB 0.9.0

### ISSUE THREAD

**Issue by nrlugg** [ibis-project/ibis#7313]
### What happened?

UDFs seem to be broken using DuckDB 0.9.0. Here's a very simple example of a broken UDF:

```python
import ibis

c = ibis.duckdb.connect()

@ibis.udf.scalar.python
def do_nothing(name: str) -> str:
    return name


c.execute(do_nothing("a"))
```

The same thing happens for PyArrow UDFs.

### What version of ibis are you using?

```
duckdb==0.9.0
ibis==7.0.0
```

### What backend(s) are you using, if any?

DuckDB

### Relevant log output

```sh
---------------------------------------------------------------------------
InvalidInputException                     Traceback (most recent call last)
/home/nathan/work/source/foundry/playground.ipynb Cell 2 line 1
      5 @ibis.udf.scalar.python
      6 def do_nothing(name: str) -> str:
      7     return name
---> 10 c.execute(do_nothing("a"))

File ~/work/source/foundry/.venv/lib/python3.9/site-packages/ibis/backends/base/sql/__init__.py:325, in BaseSQLBackend.execute(self, expr, params, limit, **kwargs)
    295 """Compile and execute an Ibis expression.
    296 
    297 Compile and execute Ibis expression using this backend client
   (...)
    319     * `Scalar`: Python scalar value
    320 """
    321 # TODO Reconsider having `kwargs` here. It's needed to support
    322 # `external_tables` in clickhouse, but better to deprecate that
    323 # feature than all this magic.
    324 # we don't want to pass `timecontext` to `raw_sql`
--> 325 self._run_pre_execute_hooks(expr)
    327 kwargs.pop("timecontext", None)
    328 query_ast = self.compiler.to_ast_ensure_limit(expr, limit, params=params)

File ~/work/source/foundry/.venv/lib/python3.9/site-packages/ibis/backends/duckdb/__init__.py:839, in Backend._run_pre_execute_hooks(self, expr)
    837     elif started is False:
    838         self._record_batch_readers_consumed[t.name] = True
--> 839 super()._run_pre_execute_hooks(expr)

File ~/work/source/foundry/.venv/lib/python3.9/site-packages/ibis/backends/base/__init__.py:943, in BaseBackend._run_pre_execute_hooks(self, expr)
    941 """Backend-specific hooks to run before an expression is executed."""
    942 self._define_udf_translation_rules(expr)
--> 943 self._register_udfs(expr)
    944 self._register_in_memory_tables(expr)

File ~/work/source/foundry/.venv/lib/python3.9/site-packages/ibis/backends/duckdb/__init__.py:1136, in Backend._register_udfs(self, expr)
   1134 registration_func = compile_func(udf_node)
   1135 if registration_func is not None:
-> 1136     registration_func(con)

File ~/work/source/foundry/.venv/lib/python3.9/site-packages/ibis/backends/duckdb/__init__.py:1145, in Backend._compile_udf.<locals>.register_udf(con)
   1144 def register_udf(con):
-> 1145     return con.connection.create_function(
   1146         name,
   1147         func,
   1148         input_types,
   1149         output_type,
   1150         type=_UDF_INPUT_TYPE_MAPPING[udf_node.__input_type__],
   1151     )

InvalidInputException: Invalid Input Error: This function can not be called with an active transaction!, commit or abort the existing one first
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by cpcloud** [comment]
Thanks for opening an issue!

This should be fixed by #7298.

**Comment by nrlugg** [comment]
Oh awesome! Thanks! I'll close this issue then.
